### PR TITLE
Libint integration ao

### PR DESCRIPTION
**Pull request by rubengerritsen**
The aomatrix part

### CODE REVIEW

File: include/votca/xtp/aobasis.h [hunk 1]
```diff
@@ -24,6 +24,15 @@
...
 #include "eigen.h"
+#include <libint2.hpp>
+
+// some versions of libint2 have no libint2::svector, in that case we typedef it
+#ifndef _libint2_include_libint2_util_smallvector_h_
+namespace libint2 {
+template <class T>
+using svector = std::vector<T>;
+}  // namespace libint2
+#endif
```
Comment (JensWehner) [inline include/votca/xtp/aobasis.h:35]: this can actually go if we only want to support libint2.6

File: include/votca/xtp/aomatrix.h [hunk 2]
```diff
@@ -23,72 +23,188 @@
 
 // Local VOTCA includes
 #include "aobasis.h"
+#include "votca/xtp/eigen.h"
+#include <libint2.hpp>
```
Comment (JensWehner) [inline include/votca/xtp/aomatrix.h:27]: not needed because aobasis includes it.

File: include/votca/xtp/aomatrix.h [hunk 3]
```diff
@@ -23,72 +23,188 @@
...
+
+  template <libint2::Operator obtype,
+            typename OperatorParams =
+                typename libint2::operator_traits<obtype>::oper_params_type>
+  std::array<MatrixLibInt, libint2::operator_traits<obtype>::nopers>
+      computeOneBodyIntegrals(const AOBasis& basis,
+                              OperatorParams oparams = OperatorParams());
+
+  std::unordered_map<Index, std::vector<Index>> compute_shellpairs(
+      const AOBasis& bs1, const double threshold = 1e-20);
```
Comment (JensWehner) [inline include/votca/xtp/aomatrix.h:53]: maybe part of aobasis?

File: src/libxtp/factories/toolfactory.cc [hunk 4]
```diff
@@ -53,6 +54,7 @@ void QMToolFactory::RegisterAll(void) {
   QMTools().Register<APDFT>("apdft");
   QMTools().Register<Mol2Orb>("mol2orb");
   QMTools().Register<Orb2Mol>("orb2mol");
+  QMTools().Register<ToLibintOrder>("tolibintorder");
 }
```
Comment (JensWehner) [inline src/libxtp/factories/toolfactory.cc:58]: We should remove this tool before we finally merge.

File: src/libxtp/molden.cc [hunk 5]
```diff
@@ -19,7 +19,7 @@ void Molden::writeAtoms(const Orbitals& orbitals,
 
 void Molden::writeMOs(const Orbitals& orbitals, std::ofstream& outFile) const {
   Eigen::VectorXd energies = orbitals.MOs().eigenvalues();
-  OrbReorder reorder(_reorderList, _multipliers);
+  OrbReorder reorder(_reorderList, _multipliers, true);
```
Comment (JensWehner) [inline src/libxtp/molden.cc:22]: the true should be split into a variable

File: src/libxtp/tools/tolibintorder.cc [hunk 6]
```diff
@@ -0,0 +1,72 @@
+#include "tolibintorder.h"
+
+// VOTCA includes
+#include <votca/tools/eigenio_matrixmarket.h>
+
+// Local VOTCA includes
```
Comment (JensWehner) [inline src/libxtp/tools/tolibintorder.cc:6]: should be removed before merge

File: src/libxtp/tools/tolibintorder.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
+
+#pragma once
+#ifndef VOTCA_XTP_TOLIBINTORDER_H
+#define VOTCA_XTP_TOLIBINTORDER_H
+
```
Comment (JensWehner) [inline src/libxtp/tools/tolibintorder.h:5]: remove file before merge

**Review by JensWehner — COMMENTED** [review]
this is a good start

**Comment by rubengerritsen** [comment]
fedora rawhide build fails with `error: 'LIBINT2_MAX_AM_default1' was not declared in this scope; did you mean 'LIBINT2_MAX_AM_default'?` The cause is known see https://github.com/evaleev/libint/issues/192, the current answer is that it will be fixed in 2.7.0, the fix is here https://github.com/evaleev/libint/pull/149

**Pull request merged**